pkg/repo/badgerdb/translation.go

```diff
@@ -20,27 +20,54 @@ func translationKey(serviceID uuid.UUID, language language.Tag) []byte {
 	return []byte(fmt.Sprintf("%s%s:%s", translationPrefix, serviceID, language))
 }
 
+func (r *Repo) Tx(ctx context.Context, fn func(repo.TranslationsRepo) error) error {
```

Comment:
Doesn't it belong to `repo.go`? `Tx` has nothing to do with the `Translation` resource.

Reply:
Hmm, got confused by the TranslationsRepo naming. Why aren't we declaring all method signatures in the Repo interface directly?

Reply:
We talk about different things. I guess.

The method `Tx()` belongs to the Repo implementation for each database - `pkg/repo/<db>/repo.go`. You put the implementation in the `pkg/repo/badgerdb/translation.go` file that MUST implement only methods for the Translation resource. Why did you put it in this file?

You talk about the interfaces in the `pkg/repo/repo.go` file where they are grouped by resource. `Tx` MUST be declared in the `Repo` interface.